lib/chasqui/subscriptions.rb

```diff
@@ -0,0 +1,67 @@
+module Chasqui
+  class Subscriptions
+    attr_reader :queue_adapter
+
+    def initialize(queue_adapter)
+      @subscriptions = {}
+      @subscribers ||= {}
+      @queue_adapter = queue_adapter
+    end
+
+    def register(subscriber)
+      subscriber.channels.each do |channel|
+        register_one channel, subscriber
+      end
+    end
+
+    def unregister(subscriber)
+      subscriber.channels.each do |channel|
+        unregister_one channel, subscriber
+      end
+    end
+
+    def find(channel, queue)
+      @subscriptions[queue.to_s][channel.to_s].values
+    end
+
+    def subscribers
+      @subscribers.values
+    end
+
+    def subscribed?(subscriber)
+      @subscribers.key? subscriber.object_id
+    end
+
+    def autoregister!
+      Subscriber.subscribers.each { |s| register s }
+    end
+
+    private
+
+    def register_one(channel, subscriber)
```

Comment:
@tpett so this method looks surprisingly close to your `.subscribe(channel, callable)` suggestion. Shouldn't be too difficult to update it to also accept a callable object in addition to a subscriber class and expose it as a public method on the Chasqui module.